Project sprint 5

MikhailFrantsuzov opened the pull request:
Логика работы приложения разбита на несколько частей.
Работа по генерации вопросов для квиза вынесена в отдельную сущность.
Написан класс для ведения статистики, которая будет учитывать лучший счёт игры в квиз среди всех игр (даже между перезапусками).
Статистика сохраняется в UserDefaults.
Вопросы не повторяются.

MikhailFrantsuzov committed:
Добавлены шрифты, цвета, картинки

MikhailFrantsuzov committed:
Добавлена копия кнопки "Да" из Figma

MikhailFrantsuzov committed:
Выполнена задача спринта 4. Экраны соответствуют макетам Figma, реализована логика приложения согласно функциональным и техническим требованиям.
Дополнительно: для кнопок "Нет" и "Да" добавлено свойство isEnabled и реализован отклик на их нажатие, путём отрисовки их рамок.

MikhailFrantsuzov committed:
1. noButton и yesButton изменены на приватные.
2. Цвет Status Bar изменён на белый.
3. Строка с private func convert разбита на несколько строк для уменьшения длины кода.

MikhailFrantsuzov committed:
Проект спринта №5.
Логика работы приложения разбита на несколько частей.
Работа по генерации вопросов для квиза вынесена в отдельную сущность.
Написан класс для ведения статистики, которая будет учитывать лучший счёт игры в квиз среди всех игр (даже между перезапусками).
Статистика сохраняется в UserDefaults.
Вопросы не повторяются.

MikhailFrantsuzov committed:
Проект спринта №6

DanilaG commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,3 +1,10 @@
 
-extension UIColor { }
+extension UIColor {
+//    static var ypGreen: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
+//    static var ypRed: UIColor { UIColor(named: "YP Red") ?? UIColor.red }
+//    static var ypBlack: UIColor { UIColor(named: "YP Black") ?? UIColor.black}
+//    static var ypBackground: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
+//    static var ypGray: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
+//    static var ypWhite: UIColor { UIColor(named: "YP White") ?? UIColor.white}
```
Давай разкомментируем

MikhailFrantsuzov committed:
Проект исправлен согласно комментариям.
1. Раскомментировал объявление цветов в extension UIColor.
2. Добавлен метод func updateAvailableQuestions для обновления массива вопросов для новой игры.
3. Скорректироан текст в алерте согласно макету.

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -9,13 +9,11 @@ final class MovieQuizViewController: UIViewController {
     private let questionsCount: Int = 10
     
-    private var questionFactory: QuestionFactory?
+    private var questionFactory: QuestionFactoryImplementation?
```
Давай вернём как было, иначе теряется смысл закрытия реализации протоколом(
Если нужно добавить новый метод, то его можно добавить в протокол и использовать через протокол

MikhailFrantsuzov replied:
Вернул

DanilaG commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -43,6 +43,9 @@ extension QuestionFactoryImplementation: QuestionFactory {
         delegate?.didReceiveQuestion(question)
     }
+    func updateAvailableQuestions() {
+            availableQuestions = questions
+        }
```
Давай поправим отступы

MikhailFrantsuzov replied:
Поправил

DanilaG commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,10 +1,10 @@
-//    static var ypBackground: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
-//    static var ypGray: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
-//    static var ypWhite: UIColor { UIColor(named: "YP White") ?? UIColor.white}
+    static var ypGreenColor: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
+    static var ypRedColor: UIColor { UIColor(named: "YP Red") ?? UIColor.red }
+    static var ypBlackColor: UIColor { UIColor(named: "YP Black") ?? UIColor.black}
+    static var ypBackgroundColor: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
+    static var ypGrayColor: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
+    static var ypWhiteColor: UIColor { UIColor(named: "YP White") ?? UIColor.white}
```
При раскоментировании у цветов добавилось слово `Color`, однако в файлах проекта оно не добавилось. Поэтому проект не может скомпилироваться

MikhailFrantsuzov replied:
Изначально я закомментил цвета потому, что так посоветовал наставник. У меня в проекте возникает конфликт между ними и цветами в ресурсах в Assets.xcassets. 
Потом я их раскомментил, но решил поменять имена, чтобы не было конфликта. 
У меня проект запускается в обоих случаях, и если их закомментить, и если поменять названия. 
Другого решения этого вопроса, кроме как удалить их либо в ресурсах, либо в расширении мне не предложили. 
Ок. Я грохнул цвета в ресурсах (надеюсь это не повлияет на проект в дальнейшем) и вернул расширение с ними к первоначальному виду. Надеюсь теперь всё будет работать без конфликтов. У меня проект запускается. 
<img width="1644" alt="Снимок экрана 2023-11-18 в 16 33 11" src="https://github.com/MikhailFrantsuzov/MovieQuiz-ios/assets/145130953/caf59dec-cc2e-457c-9324-2e14edd7c998">
